torch/_torch_docs.py

```diff
@@ -3160,6 +3160,53 @@ def merge_dicts(*dicts):
 
 """)
 
+add_docstr(torch.inner, r"""
+inner(input, other, *, out=None) -> Tensor
+
+Computes the vector inner product for 1D tensors. For higher dimensions, computes
+a contraction of :attr:`input` and :attr:`other` over their last dimension.
```

Comment:
I think this documentation is very good and we can probably ship it as-is. Maybe in the future we can think more about terms like "contraction"?